src/Sources/Code/Module.cpp

```diff
@@ -276,27 +290,49 @@ void WriteSimVar(SimVar& simVar, Client* client) {
 	);
 
 		if (hr != S_OK) {
-			fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %lu, SimVar: %s (ID: %u)", client->Name.c_str(), hr, simVar.Name.c_str(), simVar.ID);
+			fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %lu, SimVar: (ID: %u)", client->Name.c_str(), hr, simVar.ID);
 		}
 #if _DEBUG
-	std::cout << "MobiFlight[" << client->Name.c_str() << "]: Written SimVar " << simVar.Name.c_str();
+	std::cout << "MobiFlight[" << client->Name.c_str() << "]: Written SimVar";
 	std::cout << " with ID " << simVar.ID << " has value " << simVar.Value << std::endl;
 #endif
 }
 
+//check whether SimVar has already registered in SimVar list
+ReadRPNCode* IsDuplicatedSimVar(const std::string code) {
+	for (auto &rpn : RPNCodelist) {
+		if (rpn.Code == code) {
+			return &rpn;
+		}
+	}
+	return nullptr;
+}
+
 // Register a single Float-SimVar and send the current value to SimConnect Clients
 void RegisterFloatSimVar(const std::string code, Client* client) {
 	std::vector<SimVar>* SimVars = &(client->SimVars);
 	std::vector<StringSimVar>* StringSimVars = &(client->StringSimVars);
 	SimVar newSimVar;
+	ReadRPNCode* pdupRpn = IsDuplicatedSimVar(code);
```

Comment:
Please use more descriptive variable names.

```suggestion
	ReadRPNCode* DuplicatedRpnCode = GetDuplicatedRpnCodeOrNull(code);
```